cmd/kubectl-gadget/utils/trace.go

```diff
@@ -109,6 +143,375 @@ func deleteTraces(contextLogger *log.Entry, traceRestClient *restclient.RESTClie
+			}
+			labelsSelector[kv[0]] = kv[1]
+		}
+	}
+
+	namespace := ""
+	if !config.UseAllNamespaces {
+		namespace = GetNamespace()
+	}
+
+	trace := &gadgetv1alpha1.Trace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: config.GadgetName + "-",
+			Namespace:    "gadget",
+			Annotations: map[string]string{
+				GADGET_OPERATION: config.Operation,
+			},
+			Labels: map[string]string{
+				TRACE_ID: traceID,
+				/*
+				 * Add all this information here to be able to find the trace thanks
+				 * to them when calling getTraceListFromParameters().
+				 */
+				"GadgetName":    config.GadgetName,
+				"nodeName":      config.NodeName,
+				"namespace":     namespace,
+				"podName":       config.PodName,
+				"containerName": config.ContainerName,
+				"outputMode":    config.OutputMode,
+			},
+		},
+		Spec: gadgetv1alpha1.TraceSpec{
+			Node:   config.NodeName,
+			Gadget: config.GadgetName,
+			Filter: &gadgetv1alpha1.ContainerFilter{
+				Namespace:     namespace,
+				Podname:       config.PodName,
+				ContainerName: config.ContainerName,
+				Labels:        labelsSelector,
+			},
+			RunMode:    "Manual",
+			OutputMode: config.OutputMode,
+		},
+	}
+
+	err := createTraces(trace)
+	if err != nil {
+		return "", err
+	}
+
+	return traceID, nil
+}
+
+// getTraceListFromID returns an array of pointer to gadgetv1alpha1.Trace
+// corresponding to the given traceID.
+// If no trace corresponds to this ID, error is set.
+func getTraceListFromID(traceID string) (gadgetv1alpha1.TraceList, error) {
+	var listTracesOptions = metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", TRACE_ID, traceID),
+		FieldSelector: fields.Everything().String(),
+	}
+
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return gadgetv1alpha1.TraceList{}, err
+	}
+
+	var traces gadgetv1alpha1.TraceList
+
+	err = traceRestClient.
+		Get().
+		Namespace("gadget").
+		Resource("traces").
+		VersionedParams(&listTracesOptions, scheme.ParameterCodec).
+		Do(context.TODO()).
+		Into(&traces)
+	if err != nil {
+		return traces, fmt.Errorf("Error getting traces from traceID %q: %w", traceID, err)
+	}
+
+	if len(traces.Items) == 0 {
+		return traces, fmt.Errorf("No traces found for traceID %q!", traceID)
+	}
+
+	return traces, nil
+}
+
+// SetTraceOperation set the operation of an existing trace.
+// If trace does not exist or Annotations does not contain GADGET_OPERATION key
+// an error is returned.
+func SetTraceOperation(traceID string, operation string) error {
+	traces, err := getTraceListFromID(traceID)
+	if err != nil {
+		return err
+	}
+
+	for _, trace := range traces.Items {
+		err = updateTraceOperation(&trace, operation)
+		if err != nil {
+			return fmt.Errorf("error updating trace operation for %s: %w", traceID, err)
+		}
+	}
+
+	return nil
+}
+
+// PrintTraceOutput is used to print trace output.
+// Trace state must be either "Started" or "Completed" to contain some events.
+func PrintTraceOutput(traceID string, config TraceConfig,
```

Comment:
I really like this suggestion but I am not sure about the name of those two functions. I would like the name to really describe what the function does. Following my suggestions but, of course, I am open to others 🙂.

Firstly, `PrintTraceGenericOutput` is used to receive the output through a stream and we pass it a function that transforms every single JSON entry received, into the requested output mode (JSON, Columns or CustomColumns). Therefore, why don't we call it `PrintTraceOutputFromStream`? In the end, it makes only sense for the gadgets that support `Spec.OutputMode=Stream`.

On the other hand, `PrintTraceCustomOutput` needs the traces to be completed because it will print the output from the `Status.Output` of each trace. In fact, we pass it a function that iterates over a TraceList to extract the `Status.Output` of each of them. Therefore, I suggest calling it `PrintTraceOutputFromStatus`. BTW, I think it is not necessary to send the entire list of traces but just a slice with all the `Status.Output`, but that's something we can change later because it would be out of scope for this PR.

Similarly, I suggest changing `RunTraceAndPrintGenericOutput` and `RunTraceAndPrintCustomOutput` into `RunTraceAndPrintStream` and `RunTraceAndPrintStatusOutput`.

What do you think?